libavfilter/vf_lvpdnn.c

```diff
@@ -0,0 +1,468 @@
+            sws_scale(ctx->sws_grayf32_to_gray8, (const uint8_t *[4]){(const uint8_t *)dnn_output->data, 0, 0, 0},
+                      (const int[4]){frame->width * 3 * sizeof(float), 0, 0, 0},
+                      0, frame->height, (uint8_t * const*)frame->data, frame->linesize);
+
+        } else {
+            av_assert0(dnn_output->dt == DNN_UINT8);
+            av_image_copy_plane(frame->data[0], frame->linesize[0],
+                                dnn_output->data, bytewidth,
+                                bytewidth, frame->height);
+        }
+        return 0;
+    case AV_PIX_FMT_GRAY8:
+        // it is possible that data type of dnn output is float32,
+        // need to add support for such case when needed.
+        av_assert0(dnn_output->dt == DNN_UINT8);
+        av_image_copy_plane(frame->data[0], frame->linesize[0],
+                            dnn_output->data, bytewidth,
+                            bytewidth, frame->height);
+        return 0;
+    case AV_PIX_FMT_GRAYF32:
+        av_assert0(dnn_output->dt == DNN_FLOAT);
+        av_image_copy_plane(frame->data[0], frame->linesize[0],
+                            dnn_output->data, bytewidth,
+                            bytewidth, frame->height);
+        return 0;
+    case AV_PIX_FMT_YUV420P:
+    case AV_PIX_FMT_YUV422P:
+    case AV_PIX_FMT_YUV444P:
+    case AV_PIX_FMT_YUV410P:
+    case AV_PIX_FMT_YUV411P:
+        sws_scale(ctx->sws_grayf32_to_gray8, (const uint8_t *[4]){(const uint8_t *)dnn_output->data, 0, 0, 0},
+                  (const int[4]){frame->width * sizeof(float), 0, 0, 0},
+                  0, frame->height, (uint8_t * const*)frame->data, frame->linesize);
+        return 0;
+    default:
+        return AVERROR(EIO);
+    }
+
+    return 0;
+}
+
+static int filter_frame(AVFilterLink *inlink, AVFrame *in)
+{
+    char slvpinfo[256] = {0,};
+    char tokeninfo[64] = {0,};
+    AVFilterContext *context  = inlink->dst;
+    AVFilterLink *outlink = context->outputs[0];
+    LVPDnnContext *ctx = context->priv;
+    DNNReturnType dnn_result;
+    AVDictionary **metadata = &in->metadata;
+    int i, bwrite = 0;
+
+    ctx->framenum ++;
+
+    if(ctx->sample_rate > 0 && ctx->framenum % ctx->sample_rate == 0)
+    {
+        copy_from_frame_to_dnn(ctx, in);
+
+        dnn_result = (ctx->dnn_module->execute_model)(ctx->model, &ctx->output, 1);
+        if (dnn_result != DNN_SUCCESS) {
+            av_log(ctx, AV_LOG_ERROR, "failed to execute model\n");
+            av_frame_free(&in);
+            return AVERROR(EIO);
+        }
+
+        DNNData *dnn_output = &ctx->output;
+        float* pfdata = dnn_output->data;
+        int lendata = ctx->output.height;
+
+        // need all inference probability as metadata
+        for(i=0; i<lendata; i++) {
+		    snprintf(tokeninfo, sizeof(tokeninfo), "%.2f,", pfdata[i]);  
+		    strcat(slvpinfo,tokeninfo);
+            if(pfdata[i] > ctx->valid_threshold) bwrite = 1;	
```

Comment:
Do we need valid_threshold at all? Could we just write all inference results to the log file?

Reply:
Yeah, We can write all inference results. removed valid_threshold value.